test/e2e/framework/providers/aws/aws.go

```diff
@@ -196,3 +200,19 @@ func newAWSClient(zone string) *ec2.EC2 {
 	}
 	return ec2.New(session, cfg)
 }
+
+// Derives the region from a valid az name.
+// Returns an error if the az is known invalid (empty)
+func azToRegion(az string) (string, error) {
+	if len(az) < 1 {
+		return "", fmt.Errorf("invalid (empty) AZ")
+	}
+
+	r := regexp.MustCompile(`^([a-zA-Z]+-)+\d+`)
+	region := r.FindString(az)
+	if region == "" {
+		return "", fmt.Errorf("invalid AZ: %s", az)
+	}
+
+	return region, nil
+}
```

Comment:
(IMO)

This would need some test cases, and perhaps an assurance that the algorithm will remain correct given future changes that AWS could make.

Reply:
Is this still an issue?  I am not sure why will this need tests. If this code is broken, then I would expect e2es to fail.

Reply:
(I hadn't realised that this code was based on what's in _master_).